observatory-dags/observatory/dags/telescopes/oapen_metadata.py

```diff
@@ -0,0 +1,227 @@
+            'User-Agent': f'{get_ao_user_agent()}'
+        }
+        response = retry_session().get(OapenMetadataTelescope.CSV_URL, headers=headers)
+        if response.status_code == 200:
+            with open(self.csv_path, 'w') as f:
+                f.write(response.content.decode('utf-8'))
+            logging.info('Download csv successful')
+            return True
+        else:
+            raise AirflowException(f'Download csv unsuccessful, {response.text}')
+
+    def transform(self):
+        """ Transform the oapen metadata csv file by storing in a jsonl format and restructuring lists/dicts
+        :return: None
+        """
+        with open(self.csv_path, 'r') as f:
+            csv_entries = [{k: v for k, v in row.items()} for row in csv.DictReader(f, skipinitialspace=True)]
+
+        # restructure entries dict
+        nested_fields = set([key.rsplit('.', 1)[0] for key in csv_entries[0].keys() if len(key.rsplit('.', 1)) > 1])
+        list_fields = {'oapen.grant.number', 'oapen.grant.acronym', 'oapen.relation.hasChapter_dc.title',
+                       'dc.subject.classification', 'oapen.grant.program', 'oapen.redirect', 'oapen.imprint',
+                       'dc.subject.other', 'oapen.notes', 'dc.title', 'collection', 'dc.relation.ispartofseries',
+                       'BITSTREAM Download URL', 'BITSTREAM ISBN', 'oapen.collection', 'dc.contributor.author',
+                       'oapen.remark.public', 'oapen.relation.isPublisherOf', 'dc.contributor.other',
+                       'dc.contributor.editor', 'dc.relation.isreplacedbydouble', 'dc.date.issued',
+                       'dc.description.abstract', 'BITSTREAM Webshop URL', 'dc.type', 'dc.identifier.isbn',
+                       'dc.description.provenance', 'oapen.grant.project', 'oapen.relation.isbn', 'dc.identifier',
+                       'dc.date.accessioned', 'BITSTREAM License', 'oapen.relation.isFundedBy_grantor.name',
+                       'dc.relation.isnodouble', 'dc.language', 'grantor.number'}
+        entries = change_keys(csv_entries, convert, nested_fields, list_fields)
+
+        # write out the transformed data
+        with open(self.transform_path, 'w') as json_out:
+            with jsonlines.Writer(json_out) as writer:
+                for entry in entries:
+                    writer.write_all([entry])
+
+
+class OapenMetadataTelescope(StreamTelescope):
+    """ Oapen Metadata telescope """
+    DAG_ID = 'oapen_metadata'
+    CSV_URL = 'https://library.oapen.org/download-export?format=csv'
+
+    def __init__(self):
+        self.dag_id = OapenMetadataTelescope.DAG_ID
+        self.start_date = datetime(2018, 5, 14)
+        self.schedule_interval = '@weekly'
+        self.dataset_id = 'oapen'
+        self.merge_partition_field = 'id'
+        self.updated_date_field = 'dc.date.available'
+        self.bq_merge_days = 0
+
+        self.description = 'oapen metadata telescope'
+        self.schema_prefix = 'oapen_'
```

Comment:
self.dag_id is not necessary as it is set by passing a value to super().\_\_init\_\_.

All of the other attributes should be keyword arguments with default values so that they can be overridden (except perhaps self.description). They can also just be directory passed to super().\_\_init\_\_.